api-spec/api-for-io.yaml

```
@@ -0,0 +1,447 @@
+swagger: '2.0'
+info:
+  version: 1.0.0
+  title: Proxy pagoPA
+  description: Authenticated API to support payment activations on pagoPA platform
+basePath: /checkout/auth/payments/v1
+schemes:
+  - https
```

Is missing the security definition:
```suggestion
schemes:
  - https
securityDefinitions:
  apiKeyHeader:
    type: apiKey
    name: Ocp-Apim-Subscription-Key
    in: header
security:
  - apiKeyHeader: []
```

---

I defined in all api a new parameter:
`$ref: '#/parameters/SubscriptionKey'`
for client generation it is better to have this param in `securityDefinitions`?

---

Yes, parameters references are not well supported by the client generator. You can use the OpenAPI specs downloaded from the APIM (the OpenApi 2 version is available only on json format, but could be converted with [swagger validator](https://editor.swagger.io/) to the yaml version). To avoid other issues I'll even remove the `apiKeyQuery` security parameter.